src/components/ec-modal/ec-modal.spec.js

```diff
@@ -1,10 +1,19 @@
 import { mount } from '@vue/test-utils';
+import { useFocusTrap } from '@vueuse/integrations/useFocusTrap';
 import { defineComponent } from 'vue';
 
 import { withMockedConsole } from '../../../tests/utils/console';
 import EcModal from './ec-modal.vue';
 
-jest.mock('../../directives/ec-focus-trap');
+jest.mock('@vueuse/integrations/useFocusTrap', () => ({
```

Comment:
Here we are going to use in our unit test our knowledge about internal implementation that component has. It is worth testing component as a black box with interface, and do not care about code inside it.

Reply:
i don't get what i am supposed to do

Reply:
We shouldn't test the internal implementation of the component. So mocking the libraries used inside the component and not exposed outside of the component should be considered a red flag.

In this case, we should reconsider our testing approach to the trap functionality of the `ec-modal` component to avoid mocking and testing its internal implementation.

So if https://github.com/Ebury/chameleon/pull/1308/files#r987874713 would work, the mocking of useFocusTrap will not be needed.

Reply:
`useFocusTrap` is a wrapper around the `focus-trap` library and it can't run in JSDOM because it accesses lots of API that JSDOM doesn't implement. So we decided to not test the trap at all in unit tests and always ignore it. We have an integration test in EBO to check that 2 modals can be displayed at the same time, we don't have any tests checking that the focus is really trapped in a modal. If we want to test the trap, let's do it in EBO using Cypress so we have the browser environment to test everything.

We were even mocking the focus-trap directive globally using the `ec-focus-trap.mock.js` in `tests/mocks` so it is always ignored when we use our modals in unit tests.